IPV6 Exception

I'm using the Sftp.inboundAdapter and I'm getting the next exception when I configure the SFTPSessionFactory with a host in IPv6

```
Caused by: java.lang.NumberFormatException: For input string: ""
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:592)
	at java.lang.Integer.parseInt(Integer.java:615)
	at org.springframework.integration.file.remote.synchronizer.AbstractInboundFileSynchronizer.copyFileToLocalDirectory(AbstractInboundFileSynchronizer.java:499)
	at org.springframework.integration.file.remote.synchronizer.AbstractInboundFileSynchronizer.copyIfNotNull(AbstractInboundFileSynchronizer.java:400)
	at org.springframework.integration.file.remote.synchronizer.AbstractInboundFileSynchronizer.transferFilesFromRemoteToLocal(AbstractInboundFileSynchronizer.java:386)
	at org.springframework.integration.file.remote.synchronizer.AbstractInboundFileSynchronizer.lambda$synchronizeToLocalDirectory$0(AbstractInboundFileSynchronizer.java:349)
	at org.springframework.integration.file.remote.RemoteFileTemplate.execute(RemoteFileTemplate.java:439)
```

Reviewing the code I found in AbstractInboundFileSynchronizer
```
String[] hostPort = session.getHostPort().split(":");

.....

String remoteFileMetadata = (new URI(this.protocol(), (String)null, hostPort[0], Integer.parseInt(hostPort[1]), '/' + remoteDirectoryPath, (String)null, remoteFileName)).toString();
```


Probably `String[] hostPort = session.getHostPort().split(":");` can be replaced by 

```
int lio=  hostPort.lastIndexOf(":");
String host = hostPort.substring(0,lio);
String port = hostPort.substring(lio+1);
```




This is my configuration 
```
@Bean
    public IntegrationFlow readFileFlow(
        SessionFactory<LsEntry> sftpSessionFactory, IntegrationFlow moveInputFileFlow,
        IntegrationFlow processFileFlow) {
        return IntegrationFlows.from(Sftp.inboundAdapter(sftpSessionFactory)
                .preserveTimestamp(true)
                .autoCreateLocalDirectory(true)
                .deleteRemoteFiles(true)
                .maxFetchSize(1)
                .remoteDirectory("/input")
                .filter(new SftpPersistentAcceptOnceFileListFilter(new SimpleMetadataStore(), "rotate"))
                .localFilter(new AcceptOnceFileListFilter())
                .regexFilter("^([0-9\\-]+).in$")
                .localDirectory(new File("tmp")),
            e -> e.autoStartup(true)
                .poller(Pollers.fixedDelay(5000)))
            .transform(Files.toStringTransformer())
            .publishSubscribeChannel(pubSub -> pubSub
                .subscribe(moveInputFileFlow)
                .subscribe(processFileFlow))
            .get();
    }


@Bean
    public CachingSessionFactory<LsEntry> sftpSessionFactory() {
        DefaultSftpSessionFactory factory = new DefaultSftpSessionFactory(true);
        factory.setHost("::1");
        factory.setPort(2222);
        factory.setUser("foo");
        factory.setPassword("pass");
        factory.setAllowUnknownKeys(true);
        CachingSessionFactory cachingSessionFactory = new CachingSessionFactory<>(factory);
        cachingSessionFactory.setTestSession(true);
        return cachingSessionFactory;
    }
```

--- a/spring-integration-file/src/main/java/org/springframework/integration/file/remote/synchronizer/AbstractInboundFileSynchronizer.java
+++ b/spring-integration-file/src/main/java/org/springframework/integration/file/remote/synchronizer/AbstractInboundFileSynchronizer.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2002-2019 the original author or authors.
+ * Copyright 2002-2020 the original author or authors.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -493,10 +493,13 @@ public abstract class AbstractInboundFileSynchronizer<F>
 				if (this.preserveTimestamp && !localFile.setLastModified(modified)) {
 					throw new IllegalStateException("Could not sent last modified on file: " + localFile);
 				}
-				String[] hostPort = session.getHostPort().split(":");
+				String hostPort = session.getHostPort();
+				int colonIndex = hostPort.lastIndexOf(":");
+				String host = hostPort.substring(0, colonIndex);
+				String port = hostPort.substring(colonIndex + 1);
 				try {
 					String remoteFileMetadata =
-							new URI(protocol(), null, hostPort[0], Integer.parseInt(hostPort[1]),
+							new URI(protocol(), null, host, Integer.parseInt(port),
 									'/' + remoteDirectoryPath, null, remoteFileName)
 									.toString();
 					this.remoteFileMetadataStore.put(buildMetadataKey(localFile), remoteFileMetadata);

--- a/spring-integration-sftp/src/test/java/org/springframework/integration/sftp/inbound/SftpInboundRemoteFileSystemSynchronizerTests.java
+++ b/spring-integration-sftp/src/test/java/org/springframework/integration/sftp/inbound/SftpInboundRemoteFileSystemSynchronizerTests.java
@@ -17,6 +17,7 @@
 package org.springframework.integration.sftp.inbound;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.willReturn;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.times;
@@ -25,10 +26,12 @@ import static org.mockito.Mockito.when;
 
 import java.io.File;
 import java.io.FileInputStream;
+import java.net.URI;
 import java.util.ArrayList;
 import java.util.Calendar;
 import java.util.Collection;
 import java.util.List;
+import java.util.Map;
 import java.util.Vector;
 
 import org.junit.jupiter.api.AfterEach;
@@ -65,11 +68,14 @@ import com.jcraft.jsch.SftpATTRS;
  */
 public class SftpInboundRemoteFileSystemSynchronizerTests {
 
-	private static com.jcraft.jsch.Session jschSession = mock(com.jcraft.jsch.Session.class);
+	private static final com.jcraft.jsch.Session jschSession = mock(com.jcraft.jsch.Session.class);
 
 	@BeforeEach
 	@AfterEach
 	public void cleanup() {
+		willReturn("::1")
+				.given(jschSession)
+				.getHost();
 		File file = new File("test");
 		if (file.exists()) {
 			String[] files = file.list();
@@ -103,7 +109,7 @@ public class SftpInboundRemoteFileSystemSynchronizerTests {
 		List<FileListFilter<LsEntry>> filters = new ArrayList<>();
 		filters.add(persistFilter);
 		filters.add(patternFilter);
-		CompositeFileListFilter<LsEntry> filter = new CompositeFileListFilter<LsEntry>(filters);
+		CompositeFileListFilter<LsEntry> filter = new CompositeFileListFilter<>(filters);
 		synchronizer.setFilter(filter);
 		synchronizer.setBeanFactory(mock(BeanFactory.class));
 		synchronizer.afterPropertiesSet();
@@ -128,6 +134,13 @@ public class SftpInboundRemoteFileSystemSynchronizerTests {
 
 		assertThat(atestFile.getHeaders())
 				.containsKeys(FileHeaders.REMOTE_HOST_PORT, FileHeaders.REMOTE_DIRECTORY, FileHeaders.REMOTE_FILE);
+
+		@SuppressWarnings("unchecked")
+		Map<String, String> remoteFileMetadataStore =
+				TestUtils.getPropertyValue(synchronizer, "remoteFileMetadataStore.metadata", Map.class);
+
+		String next = remoteFileMetadataStore.values().iterator().next();
+		assertThat(URI.create(next).getHost()).isEqualTo("[::1]");
 
 		Message<File> btestFile = ms.receive();
 		assertThat(btestFile).isNotNull();

